tests/contrib/urllib3/test_urllib3.py

```diff
@@ -0,0 +1,392 @@
+import mock
+import pytest
+import urllib3
+
+from ddtrace import config
+from ddtrace.constants import ANALYTICS_SAMPLE_RATE_KEY
+from ddtrace.contrib.urllib3 import patch, unpatch
+from ddtrace.ext import errors, http
+
+from tests.opentracer.utils import init_tracer
+
+from ...base import BaseTracerTestCase
+
+# socket name comes from https://english.stackexchange.com/a/44048
+SOCKET = "httpbin.org"
+URL_200 = "http://{}/status/200".format(SOCKET)
+URL_500 = "http://{}/status/500".format(SOCKET)
+
+
+class BaseUrllib3TestCase(object):
+    """Provides the setup and teardown for patching/unpatching the urllib3 integration"""
+
+    def setUp(self):
+        super(BaseUrllib3TestCase, self).setUp()
+
+        patch()
+        self.http = urllib3.PoolManager()
+        setattr(urllib3.connectionpool.HTTPConnectionPool, "datadog_tracer", self.tracer)
+
+    def tearDown(self):
+        unpatch()
+
+        super(BaseUrllib3TestCase, self).tearDown()
+
+
+class TestUrllib3(BaseUrllib3TestCase, BaseTracerTestCase):
+    def test_HTTPConnectionPool_traced(self):
+        """Tests that requests made from the HTTPConnectionPool are traced"""
+        pool = urllib3.connectionpool.HTTPConnectionPool(SOCKET)
+        # Test a relative URL
+        r = pool.request("GET", "/status/200")
+        assert r.status == 200
+        assert len(self.tracer.writer.pop()) == 1
```

Comment:
Should we check what's inside the trace? At least one field to be sure it's not anything else that created it.